src/twisted/application/runner/test/mockreactor.py

```diff
@@ -0,0 +1,40 @@
+# Copyright (c) Twisted Matrix Laboratories.
+# See LICENSE for details.
+
+"""
+Mock reactor.
+"""
+
+from twisted.internet import error
+from twisted.test.proto_helpers import MemoryReactor as BaseReactor
+
+
+
+class MockReactor(BaseReactor):
```

Comment:
A weird thing about reactors is the weird use of modules as global singletons.  A reactor doesn't really have an `install` method, so that's why I didn't put it on `MemoryReactor`.  The reactor *module* has an install method, until it is replaced by a reactor, then it doesn't any more.  Or something.  It's all pretty odd.